### PR TITLE
PR login popup - correction of links

### PR DESCRIPTION
**Pull request by magelle**
After login popup merge @jdubois noticed that some links where broken.

The aim of this pull request is to fix these links.

### CODE REVIEW

File: generators/client/templates/src/main/webapp/app/components/login/login.html [hunk 1]
```diff
@@ -40,10 +40,10 @@ <h1 translate="login.title">Sign in</h1>
             </form>
             <p></p>
             <div class="alert alert-warning">
-                <a class="alert-link" href="#/reset/request" translate="login.password.forgot">Did you forget your password?</a>
+                <a class="alert-link" href="" ng-click="requestResetPassword()" translate="login.password.forgot">Did you forget your password?</a>
             </div>
-            <div class="alert alert-warning" translate="global.messages.info.register">
-                You don't have an account yet? <a class="alert-link" href="#/register">Register a new account</a>
+            <div class="alert alert-warning" translate="global.messages.info.register" translate-compile>
+                You don't have an account yet? <a class="alert-link" href="" ng-click="register()">Register a new account</a>
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/components/login/login.html]: Why did you cange to ng-click ideally this should be changed to ui-sref

File: generators/client/templates/src/main/webapp/app/home/_home.controller.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
-    .controller('HomeController', function ($scope, Principal) {
+    .controller('HomeController', function ($scope, Principal, $state, LoginService) {
         Principal.identity().then(function(account) {
             $scope.account = account;
             $scope.isAuthenticated = Principal.isAuthenticated;
         });
+
+        $scope.login = LoginService.open;
+
+        $scope.register = function () {
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/home/_home.controller.js]: why is this required, register will still work as normal link??

File: generators/client/templates/src/main/webapp/app/home/home.html [hunk 3]
```diff
@@ -12,12 +12,12 @@ <h1 translate="home.title">Welcome, Java Hipster!</h1>
...
-                    If you want to <a class="alert-link" href="#/login">sign in</a>, you can try the default accounts:<br/>- Administrator (login="admin" and password="admin") <br/>- User (login="user" and password="user").
+                <div class="alert alert-warning" ng-switch-when="false" translate="global.messages.info.authenticated" translate-compile>
+                    If you want to <a class="alert-link" href="" ng-click="login()">sign in</a>, you can try the default accounts:<br/>- Administrator (login="admin" and password="admin") <br/>- User (login="user" and password="user").
                 </div>
 
-                <div class="alert alert-warning" ng-switch-when="false" translate="global.messages.info.register">
-                    You don't have an account yet? <a class="alert-link" href="#/register">Register a new account</a>
+                <div class="alert alert-warning" ng-switch-when="false" translate="global.messages.info.register" translate-compile>
+                    You don't have an account yet? <a class="alert-link" href="" ng-click="register()">Register a new account</a>
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/home/home.html]: remove href and ng-click and replace with `ui-sref="register"`

**Comment by deepu105** [comment]
never mind the comments I have changed them and will merge

**Comment by magelle** [comment]
Actually I change the link to a ng-click because I needed to close the login modal window when the link is clicked.
Now the register link in the modal window change the state to register but the modal window stay open.

**Comment by deepu105** [comment]
Thats true only for login module in home why do you think it has to be this
way, i have removed those and it works fine
